### PR TITLE
Reporting environment issues

### PR DESCRIPTION
**Pull request by omar-selo**
## Description

Adds ability to report environment issues.

## Resolved issues

Resolves https://warthogs.atlassian.net/browse/RTW-353

## Web service API changes

Adds new endpoint `/v1/environments/reported-issues` with GET, POST, PUT, and DELETE.

## Tests

Added automated tests to the backend.

Manually tested the front-end. See video below:
[Screencast from 2024-09-17 14-05-03.webm](https://github.com/user-attachments/assets/6c5cdb5e-ab32-4a78-b495-fbd869faf2b6)

### CODE REVIEW

**Comment by andrejvelichkovski** [comment]
Just went briefly through the changes, and I noticed we don't have a `ForeignKey` relation between Environments and EnvironmentIssues. It seems that this is because we don't want to limit the report issue on one architecture, as Environments have a link to a particular architecture.

I just wanted to confirm whether this is indeed the case or there is a different reason why we are going to a manual matching.

**Comment by omar-selo** [comment]
> we don't want to limit the report issue on one architecture

I thought about this and decided it would probably make more sense for environment issues to be across architectures. So you are right.

**Review by andrejvelichkovski — APPROVED** [review]
LGTM! Thanks for this work, just left a few small style comments

File: backend/test_observer/controllers/test_cases/models.py [hunk 1]
```diff
@@ -13,8 +15,18 @@ class ReportedIssueRequest(BaseModel):
     def check_a_or_b(self):
         if not self.case_name and not self.template_id:
             raise ValueError("Either case_name or template_id is required")
+
         return self
 
+    @field_validator("url")
+    @classmethod
+    def name_must_contain_space(
```
[the remark this answers is not in the document]
Comment (omar-selo) [reply backend/test_observer/controllers/test_cases/models.py:23]: ah the typical copy paste mistakes :man_facepalming:, thanks for catching this

File: backend/tests/controllers/environments/test_environment_issues.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    _assert_reported_issue(response.json(), issue)
+
+    response = test_client.get(endpoint)
+    _assert_reported_issue(response.json()[0], issue)
+
+
+def test_mark_unconfirmed(test_client: TestClient):
+    response = test_client.post(endpoint, json=valid_post_data)
+    issue = response.json()
+    issue["is_confirmed"] = False
```
[the remark this answers is not in the document]
Comment (omar-selo) [reply backend/tests/controllers/environments/test_environment_issues.py:87]: The below call to `_assert_reported_issue(response.json(), issue)` does that already

[the remark this answers is not in the document]
Comment (andrejvelichkovski) [reply backend/tests/controllers/environments/test_environment_issues.py:87]: It seems like the `_assert_reported_issue` doesn't validate the `is_confirmed` change

```python
def _assert_reported_issue(value: dict, expected: dict) -> None:
    assert value["environment_name"] == expected["environment_name"]
    assert value["url"] == expected["url"]
    assert value["description"] == expected["description"]
    assert isinstance(value["id"], int)
    assert isinstance(value["created_at"], str)
    assert isinstance(value["updated_at"], str)
```

[the remark this answers is not in the document]
Comment (omar-selo) [reply backend/tests/controllers/environments/test_environment_issues.py:87]: You're absolutely right, I'll fix that!

**Review by andrejvelichkovski — APPROVED** [review]
Thanks for the quick follow-up, excited to see this in use!

File: frontend/lib/ui/artefact_page/environment_issues/environment_issues_preloader.dart
[the remark this answers is not in the document]
Comment (omar-selo) [reply frontend/lib/ui/artefact_page/environment_issues/environment_issues_preloader.dart:1]: Inspired by this comment, I refactored most loading of providers into two widgets: `NonBlockingProviderPreloader` and `BlockingProviderRreloader`.

File: frontend/lib/ui/artefact_page/test_event_log_expandable.dart [hunk 3]
```diff
@@ -16,66 +14,57 @@ class TestEventLogExpandable extends ConsumerWidget {
...
 
   @override
-  Widget build(BuildContext context, WidgetRef ref) {
-    final testEvents = ref.watch(testEventsProvider(testExecutionId));
-
-    return Expandable(
-      title: const Text('Event Log'),
-      initiallyExpanded: initiallyExpanded,
-      children: <Widget>[
-        testEvents.when(
```
Comment (andrejvelichkovski) [inline frontend/lib/ui/artefact_page/test_event_log_expandable.dart:26]: Aren't we deleting the Expandable here, I don't see it anywhere in the new code

**Review by andrejvelichkovski — COMMENTED** [review]
Looks good, but I think we might be breaking the event log expandable

File: frontend/lib/ui/artefact_page/test_event_log_expandable.dart (hunk 3 above)
Comment (omar-selo) [reply frontend/lib/ui/artefact_page/test_event_log_expandable.dart:26]: Great catch

Comment (omar-selo) [reply frontend/lib/ui/artefact_page/test_event_log_expandable.dart]: I tried manually testing everything now, so hopefully I didn't miss anything else

**Review by andrejvelichkovski — APPROVED** [review]
I think now we are in a good state to deploy this functionality

File: frontend/lib/ui/artefact_page/environment_issues/environment_issues_preloader.dart
[the remark this answers is not in the document]
Comment (andrejvelichkovski) [reply frontend/lib/ui/artefact_page/environment_issues/environment_issues_preloader.dart:1]: How can you not love a code clean-up :smile:

**Pull request merged**